src/modules/compliance/src/lib/FactExistenceValidator.cpp

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <FactExistenceValidator.h>
+#include <cassert>
+#include <map>
+
+namespace compliance
+{
+FactExistenceValidator::FactExistenceValidator(Behavior behavior)
+    : mBehavior(behavior)
+{
+}
+
+void FactExistenceValidator::Finish()
+{
+    if (Done())
+    {
+        return;
+    }
+
+    switch (mBehavior)
+    {
+        case Behavior::AllExist:
+            mState = Status::Compliant;
+            break;
+        case Behavior::NoneExist:
+            assert(!mHasAtLeastOneFact);
+            mState = Status::Compliant;
+            break;
+
+        case Behavior::AtLeastOneExists:
+            assert(!mHasAtLeastOneFact);
+            mState = Status::NonCompliant;
+            break;
+
+        case Behavior::AnyExist:
+            assert(!mHasAtLeastOneFact);
+            mState = Status::Compliant;
+            break;
+
+        case Behavior::OnlyOneExists:
+            mState = mHasAtLeastOneFact ? Status::Compliant : Status::NonCompliant;
+            break;
```

Comment:
>break; [](http://example.com/codeflow?start=12&length=6)

nit: break not necessary here. Q: useful to also add a default option just in case mBehavior would not be recognized (could be a bug too)